src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,67 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { Dispatch, SetStateAction, useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filter: 'all' | 'completed' | 'active',
+  setFilter: Dispatch<SetStateAction<'all' | 'completed' | 'active'>>;
+  setSearchText: Dispatch<SetStateAction<string>>,
+  searchText: string,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+// eslint-disable-next-line max-len
+export const TodoFilter: React.FC<Props> = ({
+  filter, setFilter, setSearchText, searchText,
+}) => {
+  const [handleClear, setHandleClear] = useState(false);
```

Comment:
As before, `handle` prefix is reserved for handlers, functions. Rename this variable, typically we use `is` or `has` prefixes for booleans e.g. `isCleared`  `hasTodosCleared`